src/Host.cpp

```diff
@@ -222,19 +222,85 @@ Host::~Host()
 void Host::saveModules(int sync, bool backup)
 {
     QMapIterator<QString, QStringList> it(modulesToWrite);
-    QStringList modulesToSync;
-    QString dirName = mudlet::getMudletPath(mudlet::moduleBackupsPath);
-    QDir savePath = QDir(dirName);
-    if (!savePath.exists()) {
-        savePath.mkpath(dirName);
+    mModulesToSync.clear();
+    QString savePath = mudlet::getMudletPath(mudlet::moduleBackupsPath);
+    auto savePathDir = QDir(savePath);
+    if (!savePathDir.exists()) {
+        savePathDir.mkpath(savePath);
     }
     while (it.hasNext()) {
         it.next();
         QStringList entry = it.value();
+        QString moduleName = it.key();
         QString filename_xml = entry[0];
-        // CHECKME: Consider changing datetime spec to more "sortable" "yyyy-MM-dd#HH-mm-ss" (1 of 6)
-        QString time = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
+
+        if (backup) {
+            // CHECKME: Consider changing datetime spec to more "sortable" "yyyy-MM-dd#HH-mm-ss" (1 of 6)
+            QString time = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
+            savePathDir.rename(filename_xml, savePath + moduleName + time); //move the old file, use the key (module name) as the file
+        }
+
+        auto writer = new XMLexport(this);
+        writers.insert(filename_xml, writer);
+        writer->writeModuleXML(moduleName, filename_xml);
+
+        if (entry[1].toInt()) {
+            mModulesToSync << moduleName;
+        }
+    }
+    modulesToWrite.clear();
+
+    if (sync) {
+        connect(this, &Host::profileSaveFinished, this, &Host::slot_reloadModules);
+    }
+}
+
+void Host::slot_reloadModules()
+{
+    //synchronize modules across sessions
+    QMap<Host*, TConsole*> activeSessions = mudlet::self()->mConsoleMap;
+    QMapIterator<Host*, TConsole*> sessionIterator(activeSessions);
+    while (sessionIterator.hasNext()) {
+        sessionIterator.next();
+        Host* host = sessionIterator.key();
+        if (host->mHostName == mHostName) {
+            continue;
+        }
+        QMap<QString, int> modulePri = host->mModulePriorities;
+        QMap<int, QStringList> moduleOrder;
+        for (auto it = modulePri.keyBegin(); it != modulePri.keyEnd(); ++it) {
+            moduleOrder[modulePri[*it]].append(*it);
+        }
+        QMapIterator<int, QStringList> it(moduleOrder);
+        while (it.hasNext()) {
+            it.next();
+            QStringList moduleList = it.value();
+            for (int i = 0, total = moduleList.size(); i < total; ++i) {
+                QString moduleName = moduleList[i];
+                if (mModulesToSync.contains(moduleName)) {
+                    host->reloadModule(moduleName);
```

Comment:
:thinking: It would probably be a good idea to make the module reloading a slot that can be invoked by the `mudlet` singleton with a `QStringList` of absolute module paths to be reloaded for all the profiles that are not this one (have to get the paths because the name itself is not enough if the Hosts are using *different physical files* which just happen to have the same *name*)!

Doing it *that* way, the working out whether a module needs to be reloaded can be left to the `Host` instance that gets the signal - which is altogether better for Host independence/sand-boxing rather than have one instance looking around in the innards of the others.  Of course, you are then getting in the area of `QFutures` if you *really* do not want this Host instance to continue until all the other Hosts *have* reloaded the updated module(s). Realistically though, if we ever make proper use of multi-threading, we will want the different Host instances to be running in separate threads/on separate processor and then examining each others variables becomes a whole lot more unsafe to do - so halting this one whilst the others update themselves is not optimum IMHO.